jerry-core/ecma/base/ecma-globals.h

```diff
@@ -164,6 +164,15 @@ typedef int32_t ecma_integer_value_t;
   (ECMA_INTEGER_NUMBER_MIN <= (num) && (num) <= ECMA_INTEGER_NUMBER_MAX)
 
 /**
+ * Maximum integer number for multpily two ecma values
+ */
+#if CONFIG_ECMA_NUMBER_TYPE == CONFIG_ECMA_NUMBER_FLOAT32
+#define ECMA_INTEGER_MULTIPLY_MAX 0xb50
```

Comment:
I just realize this conditional here. Why is this dependent on float size? The only constraint for `ECMA_INTEGER_MULTIPLY_MAX` that I see in the code is that it should be (less than) the square root of `ECMA_INTEGER_NUMBER_MAX` (rounded down). This should be the same value both for float32 and float64, IMHO.
PS: I also suggest to adapt de description. Something like: "Maximum integer number, which if squared, still fits in ecma_integer_value_t"